### PR TITLE
Return masthead partial

### PR DESCRIPTION
**Pull request by alkeshrails**
https://aicrowd.monday.com/boards/363093650/pulses/522766588
Expose challenge Masterhead as API

### CODE REVIEW

File: app/controllers/api/v1/challenges/challenges_controller.rb [hunk 1]
```diff
@@ -30,12 +39,44 @@ def update
           end
         end
 
+        def masthead
```
Comment (piotrekpasciak) [inline app/controllers/api/v1/challenges/challenges_controller.rb:42]: We could use some basic specs for this endpoint.

File: app/controllers/api/v1/challenges/challenges_controller.rb [hunk 2]
```diff
@@ -30,12 +39,44 @@ def update
...
+          if @challenge.present?
```
Comment (piotrekpasciak) [inline app/controllers/api/v1/challenges/challenges_controller.rb:43]: This if statement looks to be redundant and you're never going to reach else condition.

Isn't `Challenge.friendly.find(params[:id])` going to raise 404 error when challenge not presennt?

File: app/controllers/api/v1/challenges/challenges_controller.rb [hunk 3]
```diff
@@ -30,11 +39,43 @@ def update
...
+            render json: { error: 'Challenge not present.' }, status: :unprocessable_entity
+          end
+        end
+
         private
 
         def set_challenge
-          @challenge = Challenge.friendly.find(params[:id])
-          authorize @challenge, :edit?
+          @challenge = Challenge.find_by_slug(params[:id])
```
Comment (piotrekpasciak) [inline app/controllers/api/v1/challenges/challenges_controller.rb:61]: Previous version was mutch better: 
`Challenge.friendly.find(params[:id])`

File: app/controllers/api/v1/challenges/challenges_controller.rb [hunk 4]
```diff
@@ -30,11 +39,43 @@ def update
...
+            render json: {
+              masthead:
+                render_to_string(
+                  partial: 'shared/challenges/masthead',
+                  formats: :html,
+                  layout: false,
+                  locals: { challenge: @challenge, challenge_rounds: @challenge_rounds, vote: @vote, follow: @follow }
+                )
+            }
+          else
```
Comment (piotrekpasciak) [inline app/controllers/api/v1/challenges/challenges_controller.rb:53]: Get rid of this `if else` block after you'll use 
`Challenge.friendly.find(params[:id])`

File: app/controllers/api/v1/challenges/challenges_controller.rb [hunk 5]
```diff
@@ -2,8 +2,17 @@ module Api
   module V1
     module Challenges
       class ChallengesController < ::Api::V1::BaseController
-        before_action :auth_by_participant_api_key, only: [:create, :update]
-        before_action :set_challenge, only: :update
+        include ActionController::Helpers
```
Comment (piotrekpasciak) [inline app/controllers/api/v1/challenges/challenges_controller.rb:5]: What kind of `ActionController::Helpers` do we need here?

Comment (alkeshrails) [reply app/controllers/api/v1/challenges/challenges_controller.rb:5]: partial `shared/challenges/masthead` need, below listed helper ChallengesHelper, VotesHelper, FollowsHelper. so use that helper I need to `include ActionController::Helpers`

Comment (piotrekpasciak) [reply app/controllers/api/v1/challenges/challenges_controller.rb:5]: Alright, I thought including only our helpers would be sufficient. Looks good then.

File: app/controllers/api/v1/challenges/challenges_controller.rb [hunk 6]
```diff
@@ -30,11 +39,39 @@ def update
...
+              )
+          }
+        end
+
         private
 
         def set_challenge
           @challenge = Challenge.friendly.find(params[:id])
-          authorize @challenge, :edit?
+          authorize @challenge, :edit? if @challenge
```
Comment (piotrekpasciak) [inline app/controllers/api/v1/challenges/challenges_controller.rb:58]: We should get rid of `if @challenge` here and setters below.

**Pull request merged**